app/src/main/java/org/schabi/newpipe/fragments/list/search/filter/BaseSearchFilterDialogFragment.java

```diff
@@ -0,0 +1,188 @@
+                                        ViewGroup container);
+
+    @Override
+    public View onCreateView(@NonNull final LayoutInflater inflater,
+                             final ViewGroup container,
+                             final Bundle savedInstanceState) {
+        final View rootView = getRootView(inflater, container);
+        initializeFilterData();
+        return rootView;
+    }
+
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        dialogGenerator.onResume();
+    }
+
+    @Override
+    public void onStop() {
+        dialogGenerator.onPause();
+        super.onStop();
+    }
+
+    @Override
+    public void onViewCreated(@NonNull final View view, final Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        Icepick.restoreInstanceState(this, savedInstanceState);
+
+        initToolbar(getToolbar());
+    }
+
+    protected void initToolbar(final Toolbar toolbar) {
+        toolbar.setTitle(R.string.filter);
+        toolbar.setNavigationIcon(R.drawable.ic_arrow_back);
+        toolbar.inflateMenu(R.menu.menu_search_filter_dialog_fragment);
+        toolbar.setNavigationOnClickListener(v -> dismiss());
+        toolbar.setNavigationContentDescription(R.string.cancel);
+
+        final View okButton = toolbar.findViewById(R.id.search);
+        okButton.setEnabled(true);
+
+        final View resetButton = toolbar.findViewById(R.id.reset);
+        resetButton.setEnabled(true);
+
+        toolbar.setOnMenuItemClickListener(item -> {
+            if (item.getItemId() == R.id.search) {
+                dialogGenerator.prepareForSearch();
+                return true;
+            } else if (item.getItemId() == R.id.reset) {
+                dialogGenerator.reset();
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull final Bundle outState) {
+        super.onSaveInstanceState(outState);
+        // get data to save its state via Icepick
+        userSelectedContentFilterList = dialogGenerator.getSelectedContentFilters();
+        userSelectedSortFilterList = dialogGenerator.getSelectedSortFilters();
+
+        Icepick.saveInstanceState(this, outState);
+    }
+
+    private void sendDataToParentFragment() {
+        final Listener listener = (Listener) getTargetFragment();
+        if (listener != null) {
+            listener.onFinishSearchFilterDialog(
+                    userSelectedContentFilterList, userSelectedSortFilterList,
+                    selectedContentFilters, selectedSortFilters);
+        }
+        dismiss();
+    }
+
+    /**
+     * Listener to be implemented by the parent Fragment so it can receive data.
+     */
+    public interface Listener {
+
+        void onFinishSearchFilterDialog(ArrayList<Integer> userSelectedContentFilterList,
+                                        ArrayList<Integer> userSelectedSortFilterList,
```

Comment:
See the earlier suggestion.

```suggestion
        void onFinishSearchFilterDialog(List<Integer> userSelectedContentFilterList,
                                        List<Integer> userSelectedSortFilterList,
```